pandas/core/indexes/base.py

```diff
@@ -473,6 +473,28 @@ def _simple_new(cls, values, name: Label = None):
     def _constructor(self):
         return type(self)
 
+    def _maybe_check_unique(self):
+        from pandas.errors import DuplicateLabelError
+
+        if not self.is_unique:
+            # TODO: position, value, not too large.
+            msg = """Index has duplicates."""
+            duplicates = self._format_duplicate_message()
```

Comment:
instead of adding this, can you not just repr the index where we *already* handle this correctly?

Reply:
Can you clarify? You're saying we already have a method to build a DataFrame displaying duplicates and their locations?